lib/pat.c

```diff
@@ -6001,7 +6008,8 @@ grn_pat_defrag(grn_ctx *ctx, grn_pat *pat)
   uint32_t n_records = grn_pat_size(ctx, pat);
   if (n_records == 0) {
     reduced_bytes = pat->header->curr_key;
-    pat->header->curr_key = 0;
+    pat_update_curr_key(ctx, pat, 0);
```

Comment:
Not related to the main topic, but I changed it because it is better to update with pat_update_curr_key().